community/kernel/src/test/java/org/neo4j/kernel/GraphDatabaseFacadeFactoryTest.java

```diff
@@ -0,0 +1,126 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.RETURNS_MOCKS;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Map;
+
+public class GraphDatabaseFacadeFactoryTest
+{
+    @Rule
+    public final TargetDirectory.TestDirectory dir = TargetDirectory.testDirForTest( getClass() );
+
+    @Test
+    public void shouldThrowAppropriateExceptionIfStartFails()
+    {
+        // Given
+        RuntimeException startupError = new RuntimeException();
+
+        GraphDatabaseFacadeFactory db = newFaultyGraphDatabaseFacadeFactory( startupError );
+        GraphDatabaseFacade mockFacade = mock( GraphDatabaseFacade.class );
+        try
+        {
+            // When
+            db.newFacade( Collections.<String, String>emptyMap(), mock( GraphDatabaseFacadeFactory.Dependencies.class ),
+                    mockFacade );
+            fail( "Should have thrown " + RuntimeException.class );
+        }
+        catch ( RuntimeException exception )
+        {
+            // Then
+            assertEquals( startupError, Exceptions.rootCause( exception ) );
+        }
+    }
+
+    @Test
+    public void shouldThrowAppropriateExceptionIfBothStartAndShutdownFail()
+    {
+        // Given
+        RuntimeException startupError = new RuntimeException();
+        RuntimeException shutdownError = new RuntimeException();
+
+        GraphDatabaseFacadeFactory db = newFaultyGraphDatabaseFacadeFactory( startupError );
+        GraphDatabaseFacade mockFacade = mock( GraphDatabaseFacade.class );
+        doThrow( shutdownError ).when( mockFacade ).shutdown();
+        try
+        {
+            // When
+            db.newFacade( Collections.<String, String>emptyMap(), mock( GraphDatabaseFacadeFactory.Dependencies.class ),
+                    mockFacade );
+            fail( "Should have thrown " + RuntimeException.class );
+        }
+        catch ( RuntimeException exception )
+        {
+            // Then
+            assertEquals( shutdownError, exception );
+            assertEquals( startupError, Exceptions.rootCause( exception.getSuppressed()[0] ) );
+        }
+    }
+
+    private GraphDatabaseFacadeFactory newFaultyGraphDatabaseFacadeFactory( final RuntimeException startupError )
+    {
+        return new GraphDatabaseFacadeFactory()
+        {
+            @Override
+            protected PlatformModule createPlatform( Map<String, String> params, Dependencies dependencies,
+                                                     GraphDatabaseFacade graphDatabaseFacade )
+            {
+                PlatformModule module = new PlatformModule( mock( Map.class ), mock( Dependencies.class, RETURNS_MOCKS ),
+                        mock( GraphDatabaseFacade.class ) );
+                module.life = mock( LifeSupport.class );
+                module.storeDir = mock( File.class );
```

Comment:
It would be better to pass in mocks during construction, rather than poking them into what are essentially internal properties.